geocode-glib: drop, geocode-glib_2: refactor

LordGrimmauld opened the pull request:
libsoup_3 support was added upstream in release 3.26.3 and and stabilized in 3.26.4, see upstream commit aab9d79aff2424ad5ee74a86c8b51166cb6dc4dd.

The only dependent in nixpkgs already uses libsoup_3, see https://github.com/NixOS/nixpkgs/pull/388415.

`geocode-glib` in tuner was unused and can be dropped.

<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

jtojnar commented:
This is yet another package that has `libsoup3` variant (under `geocode-glib_2` attribute).

LordGrimmauld commented:
Ah damn it! I hate all-packages overrides like that... I'll just rename the `geocode-glib_2` version to `geocode-glib` and remove the libsoup2 one, there is absolutely no reason to keep that anymore. No uses in nixpkgs.

jtojnar commented:
:+1: The only use `tuner` looks like it does not actually use it any more: https://github.com/louis77/tuner/commit/04b6e1a5dc8a141bc6b863b641ae8f92c0f46681

LordGrimmauld commented:
Amazing... Alright, removing from there

jtojnar commented:
Though I would keep the `geocode-glib_2` name, no need to rename it when it is functionally different package from `geocode-glib-1.0.pc`.

LordGrimmauld commented:
Ok fair

LordGrimmauld committed:
tuner: remove unused dependencies

geoclue and geocode were removed upstream in https://github.com/louis77/tuner/commit/04b6e1a5dc8a141bc6b863b641ae8f92c0f46681

LordGrimmauld committed:
geocode-glib: drop, geocode-glib_2: refactor

LordGrimmauld commented:
Ah hold on i forgot to fix the passthru test...

LordGrimmauld committed:
nixos/tests/geocode-glib: point to geocode-glib_2

jtojnar commented:
Could you please move the tests commit before the dropping, so it continues to eval?

nyabinary commented on pkgs/by-name/ge/geocode-glib_2/package.nix:
```diff
@@ -53,28 +53,30 @@ stdenv.mkDerivation rec {
     };
   };
 
   meta = with lib; {
```
Can we remove the ``with lib`` usage here?

nyabinary commented on pkgs/by-name/ge/geocode-glib_2/package.nix:
```diff
@@ -53,28 +53,30 @@ stdenv.mkDerivation rec {
   passthru = {
     updateScript = gnome.updateScript {
-      packageName = pname;
+      packageName = finalAttrs.pname;
```
`pname` is bad practice. Use the package name directly instead.

nyabinary replied:
(Also can we remove the ``with lib`` usage in meta here?)

LordGrimmauld replied:
checking recent changes it seems the pname things are indeed being replaced, though i don't actually understand why.

nyabinary reviewed (changes requested):
Just a few things here :P

jtojnar reviewed (approved):
LGTM, thanks.
